cdk/v2/pinterest_tag/procWorkflow.yaml

```diff
@@ -0,0 +1,209 @@
+            " pair of client_ip_address and client_user_agent.")
+      )
+  - name: ecomFields
+    condition: message.type = $EventType.TRACK
+    steps:
+      - name: customFields
+        template: |
+          (
+            $customFields := message.({
+              "currency": properties.currency,
+              "value": properties.[value, total, revenue][0],
+              "num_items": $number(properties.numOfItems),
+              "order_id": properties.order_id,
+              "search_string": properties.query
+            });
+          )
+      - name: singleProductFields
+        condition: $not($exists(message.properties.products))
+        template: |
+          message.properties.({
+            "num_items": $number(quantity),
+            "content_ids": [[product_id, sku, id][0]],
+            "contents": [{
+              "quantity": [$number(quantity), 1][0],
+              "item_price": $string(price)
+            }]
+          })
+      - name: multiProductsFields
+        condition: $exists(message.properties.products)
+        template: |
+          {
+            "num_items": $sum(message.properties.products.quantity.$number()),
+            "content_ids": message.properties.products.[product_id, sku, id][0],
+            "contents": message.properties.products.{
+              "quantity": [$number(quantity), %.quantity, 1][0],
+              "item_price": $string([price, %.price][0])
+            }
+          }
+      - name: combineAllEcomFields
+        template: |
+          $outputs.ecomFields.(
+            $merge([
+              singleProductFields, 
+              multiProductsFields,
+              customFields
+            ])
+          )
+  - name: basePayload
+    template: |
+      (
+        $payload := $outputs.(
+          $merge([
+            commonFields, 
+            {"user_data": userFields}, 
+            {"custom_data": ecomFields}
+          ])
+        );
+        $sift($payload, function($v) {(
+          $length($string($v)) != 0
+        )})
+      )
+  - name: eventNames
+    steps:
+      - name: eventNamesForPage
+        description: Computes event name for page message type
+        condition: message.type = $EventType.PAGE
+        template: |
+          ($exists(message.category) ? [ "ViewCategory" ] : [ "PageVisit" ])
+      - name: eventNamesForOthers
+        description: Computes event name for other message types
+        condition: message.type != $EventType.PAGE
+        template: |
+          (
+            $event := message.[event, name][0];
+            $eventMap := destination.Config.eventsMapping.{from: to};
+            $eventNames := [
+              $lookup($eventMap, $event), 
+              $ecomEventMaps[$lowercase($event) in src].dest
+            ];
+            $count($eventNames) = 0 ? ["custom"] : $eventNames;
+          )
+  - name: destEvents
+    template: |
+      (
+        $map($outputs.eventNames, function($eventName){(
```

Comment:
as $eventNames contains combination of destination.Config.eventsMapping + ecomEventMap we will be multiplexing all.
The logic would be 
if destination.Config.eventsMapping -> multiplex deliver based on number of common mappings
else ecomEventMaps -> deliver based on this mapping

Reply:
Got it.